src/radiation/source_terms_multi_group.hpp

```diff
@@ -560,6 +633,32 @@ void RadSystem<problem_t>::AddSourceTermsMultiGroup(array_t &consVar, arrayconst
 
 			if constexpr (gamma_ != 1.0) {
 
+				// Step 1.2.0: If enable_dust_model, determine if or not to use the weak-coupling approximation
+
+				int dust_model = 0;
+				double T_d0 = NAN;
+				double lambda_gd_times_dt = NAN;
+				if constexpr (gamma_ != 1.0) {
+					if constexpr (enable_dust_gas_thermal_coupling_model_) {
+						const double T_gas0 = quokka::EOS<problem_t>::ComputeTgasFromEint(rho, Egas0, massScalars);
+						AMREX_ASSERT(T_gas0 >= 0.);
+						T_d0 = ComputeDustTemperatureBateKeto(T_gas0, T_gas0, rho, Erad0Vec, coeff_n, dt, NAN, 0, radBoundaries_g_copy);
+						AMREX_ASSERT_WITH_MESSAGE(T_d0 >= 0., "Dust temperature is negative!");
+						if (T_d0 < 0.0) {
+							amrex::Gpu::Atomic::Add(&p_iteration_failure_counter[1], 1);
+						}
+
+						const double max_Gamma_gd = coeff_n * std::max(std::sqrt(T_gas0) * T_gas0, std::sqrt(T_d0) * T_d0);
+						const double convergence_tol_for_dust_gas_coupling = 1.0e-6;
```

Comment:
How did you determine this value?

Reply:
I wanted to set that as a free parameter, but that would introduce another argument to `AddSourceTermsMultiGroup`. Is there a way to change the value of this variable in problem.cpp without having to pass it to `AddSourceTermsMultiGroup`?

Reply:
You could make it a static member variable of RadiationSystem. This may or may not be the best long-term solution, but it should work for now.

Reply:
Actually, you could make it part of RadSystem_Traits, and just access it directly.

Reply:
Fine. Instead of using `RadSystem_Traits`, I'll create a new trait, say `ISM_Traits`, so that I don't have to modify any of the tests that don't use the dust model. I'll add more parameters to `ISM_Traits` when I introduce line cooling/heating. How does that sound?